### PR TITLE
LPD-36744 add external-reference-code to the Repository entity on service

### PR DESCRIPTION
**Pull request by AliciaGarciaGarcia**
LPD-36744 add external-reference-code to the Repository entity

# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-36744

As a developer I want Repositories to use ERC so that the data referenced is migrated properly in import/export scenarios

# How am I fixing it?

The repository table should have a column for externalReferenceCode so I adding it to it.

# How can you verify that it works?

Run the included automated tests

# Commits

- **LPD-36744 add external-reference-code to the repository entity on service**
- **LPD-36744 buildService**
- **LPD-36744 baseline**
- **LPD-36744 on addRepository method assert that ERC is the same that UUID**
- **LPD-36744 upgrade process for Repository table**
- **LPD-36744 test for upgrade process**

### CODE REVIEW

File: portal-impl/src/com/liferay/portal/service/impl/RepositoryLocalServiceImpl.java [hunk 1]
```diff
@@ -57,7 +57,11 @@ public Repository addRepository(
 
 		Repository repository = repositoryPersistence.create(repositoryId);
 
-		repository.setUuid(serviceContext.getUuid());
+		String uuid = serviceContext.getUuid();
+
+		repository.setUuid(uuid);
+		repository.setExternalReferenceCode(uuid);
```
Comment (adolfopa) [inline portal-impl/src/com/liferay/portal/service/impl/RepositoryLocalServiceImpl.java:63]: I think it is not necessary to set the external reference code in this case. See `RepositoryPersistenceImpl` around line 2859 in this same PR (in the autogen commit).

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:content-management

**Review by adolfopa — APPROVED** [review]

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade

**Comment by AliciaGarciaGarcia** [comment]
ci:report:22523702

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1347401_6778

**Comment by AliciaGarciaGarcia** [comment]
ci:report:23224406

**Comment by AliciaGarciaGarcia** [comment]
rebased

**Comment by AliciaGarciaGarcia** [comment]
ci:test:relevant

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:stable

**Comment by AliciaGarciaGarcia** [comment]
There is changes on 5d834343440f8051b8853a9b1456e713695e3a43 so I should rebase and change the test

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:sf

**Comment by AliciaGarciaGarcia** [comment]
ci:test:stable

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:test:upgrade

**Comment by AliciaGarciaGarcia** [comment]
ci:report:26964965

**Comment by AliciaGarciaGarcia** [comment]
ci:reevaluate:1277401_12653

**Comment by AliciaGarciaGarcia** [comment]
ci:report:27110207

**Comment by AliciaGarciaGarcia** [comment]
master rebase conflicts

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:report:28174766

**Comment by AliciaGarciaGarcia** [comment]
ci:forward:force

**Pull request closed without merging**